tianshou/data/batch.py

```diff
@@ -511,13 +531,35 @@ def stack_(self,
 
     @staticmethod
     def stack(batches: List['Batch'], axis: int = 0) -> 'Batch':
-        """Stack a :class:`~tianshou.data.Batch` object into a
-        single new batch.
+        """Stack a :class:`~tianshou.data.Batch` object into a single new
+        batch.
         """
         batch = Batch()
         batch.stack_(batches, axis)
         return batch
 
+    def empty_(self) -> None:
+        """Empty a :class:`~tianshou.data.Batch` object with 0 or ``None``
+        filled.
+        """
+        for k, v in self.items():
+            if isinstance(v, np.ndarray) and v.dtype == np.object:
+                self.__dict__[k].fill(None)
```

Comment:
I've not got it. Could you give an example?

Reply:
It fails if `isinstance(dtype, (np.integer, int))`, for example: `np.array([1]).fill(None)`

I think it is more practical to use None only for object array, since it is usually more appropriate to have 0 for unknown data. For instance, if you do averaging of info data, missing data must be ignored, and using 0.0 is handy instead of having explicitly skip `nan` values.
```
self.__dict__[k].fill(None if self.__dict__[k].dtype == np.object else 0)
```

Reply:
But np.array([1]) is `dtype==np.int64` instead of np.object, which will not get into line 547.

Reply:
YEs you are right, I missed that point somehow x)

Reply:
Yet, you could have used `fill` for any numpy array using `self.__dict__[k].fill(None if self.__dict__[k].dtype == np.object else 0)`, which is a bit clearer than `* 0` for me, but as you wish !

Reply:
But *0 can cause another issue: if a==inf or a==nan, *0 will still be nan instead of 0.

Reply:
So are you fixing line 557 in accordance then ?

Reply:
Yes, but I found it is non-trivial: how can we determine whether a np.array has flexible type?

Reply:
What do you mean by "flexible" type ? `np.object` is for objects, `np.inexact` is for floats, and `np.integer` is for integer. For me, we should use None only for `np.object`, 0 otherwise.